plugins/chat/app/models/chat/channel.rb

```diff
@@ -107,6 +107,18 @@ def self.ensure_consistency!
       update_counts
     end
 
+    def can_be_joined_by?(user)
+      user.guardian.can_chat? && user.guardian.can_join_chat_channel?(self)
+    end
+
+    def joined_by?(user)
+      user.user_chat_channel_memberships.any? do |membership|
```

Comment:
maybe we add a `strict_loading` here?

it has huge potential for N+1 otherwise

Reply:
Makes sense 👍.